src/Diagnostics.RuntimeHost/Services/SourceWatcher/Watchers/StorageWatcher.cs

```diff
@@ -69,14 +120,82 @@ public virtual async Task CreateOrUpdatePackage(Package pkg)
 
         public void Start()
         {
-            throw new NotImplementedException();
+            blobDowloadTask = StartBlobDownload(true);
+            StartPollingForChanges();
         }
 
-        public  Task WaitForFirstCompletion()
+        private async void StartPollingForChanges()
         {
-            throw new NotImplementedException();
+            await blobDowloadTask;
+            DiagnosticsETWProvider.Instance.LogAzureStorageMessage(nameof(StorageWatcher), $"Start up blob download task completed at {DateTime.UtcNow}");
+            cacheLastModifiedTime = DateTime.UtcNow;
+            do
+            {
+                await Task.Delay(_pollingIntervalInSeconds * 1000);
+                await StartBlobDownload(false);
+                DiagnosticsETWProvider.Instance.LogAzureStorageMessage(nameof(StorageWatcher), $"Polling for blob download task completed at {DateTime.UtcNow}");
+                cacheLastModifiedTime = DateTime.UtcNow;
+            } while (true);
         }
 
+       
+        private async Task StartBlobDownload(bool startup = false)
+        {
+            var detectorsList = await storageService.GetEntitiesByPartitionkey("Detector");
+            var gists = await storageService.GetEntitiesByPartitionkey("Gist");
+            var entitiesToLoad = new List<DiagEntity>();
+            var filteredDetectors = LoadOnlyPublicDetectors ? detectorsList.Where(row => !row.IsInternal).ToList() : detectorsList.Where(row => row.IsInternal);
```

Comment:
should this be :
 var filteredDetectors = LoadOnlyPublicDetectors ? detectorsList.Where(row => !row.IsInternal).ToList() : **detectorsList**;

Reply:
You are right.